tests/languages.test.ts

```diff
@@ -83,76 +114,140 @@ describe('Che-Java sample tests on Quarkus Project', () => {
+            );
             assert.equal(formattingEdits[1].newText, "\n  ");
-            assert.deepEqual(formattingEdits[1].range, new vscode.Range(new vscode.Position(11, 9), new vscode.Position(12, 5)));
+            assert.deepEqual(
+                reviveRange(formattingEdits[1].range),
+                new theia.Range(
+                    new theia.Position(10, 8),
+                    new theia.Position(11, 4)
+                )
+            );
             assert.equal(formattingEdits[2].newText, "\n  ");
-            assert.deepEqual(formattingEdits[2].range, new vscode.Range(new vscode.Position(12, 36), new vscode.Position(13, 5)));
+            assert.deepEqual(
+                reviveRange(formattingEdits[2].range),
+                new theia.Range(
+                    new theia.Position(11, 35),
+                    new theia.Position(12, 4)
+                )
+            );
             assert.equal(formattingEdits[3].newText, "\n\n    ");
-            assert.deepEqual(formattingEdits[3].range, new vscode.Range(new vscode.Position(13, 28), new vscode.Position(15, 9)));
+            assert.deepEqual(
+                reviveRange(formattingEdits[3].range),
+                new theia.Range(
+                    new theia.Position(12, 27),
+                    new theia.Position(14, 8)
+                )
+            );
             assert.equal(formattingEdits[4].newText, "\n    ");
-            assert.deepEqual(formattingEdits[4].range, new vscode.Range(new vscode.Position(15, 46), new vscode.Position(16, 9)));
+            assert.deepEqual(
+                reviveRange(formattingEdits[4].range),
+                new theia.Range(
+                    new theia.Position(14, 45),
+                    new theia.Position(15, 8)
+                )
+            );
             assert.equal(formattingEdits[5].newText, "\n  ");
-            assert.deepEqual(formattingEdits[5].range, new vscode.Range(new vscode.Position(16, 31), new vscode.Position(17, 5)));
+            assert.deepEqual(
+                helper.reviveRange(formattingEdits[5].range),
+                new theia.Range(
+                    new theia.Position(15, 30),
+                    new theia.Position(16, 4)
+                )
+            );
+        } else {
+            assert.fail();
         }
     });
 
     it('Test Java range formatting on basic java file', async () => {
-        await vscode.window.showTextDocument(mySampleURI);
-        const formattingEdits = await vscode.commands.executeCommand("vscode.executeFormatRangeProvider", mySampleURI, new Range(new Position(1, 1), new Position(10, 25)), {
-            insertSpaces: true,
-            tabSize: 2
-        } as FormattingOptions) as TextEdit[];
-        if (!formattingEdits) {
-            assert.fail("Expected formatting edits");
-        } else {
+        await theia.window.showTextDocument(mySampleURI);
+        const formattingEdits = await che.languages.test.documentRangeFormattingEdits(
```

Comment:
we can't return directly the 'correct object' without having to 'revive' it ?

Reply:
the objects are correct, it was wrong way to compare them. Fixed